ultradns-rest/src/main/java/denominator/ultradns/UltraDNSRestGeoSupport.java

```diff
@@ -0,0 +1,242 @@
+    Collection<Region> topLevelRegions = response.iterator().next();
+    TreeSet<String> sortedEffectiveCodes = getSortedEffectiveCodes(topLevelRegions);
+    String commaSeparatedCodes = getCommaSeparatedEffectiveCodes(sortedEffectiveCodes);
+
+    response = api2.getAvailableRegions(commaSeparatedCodes);
+    Iterator<Collection<Region>> responseIterator = response.iterator();
+    for (String effectiveCode : sortedEffectiveCodes) {
+      Region region1 = regionGiven(effectiveCode, topLevelRegions);
+      Collection<Region> secondLevelRegions = responseIterator.next();
+      region1.setChildRegions(secondLevelRegions);
+    }
+
+    TreeSet<Region> sortedSecondLevelRegions = getSortedSecondLevelRegions(topLevelRegions);
+    sortedEffectiveCodes = getSortedEffectiveCodes(sortedSecondLevelRegions);
+    String[] sortedEffectiveCodesArr = new String[sortedEffectiveCodes.size()];
+    sortedEffectiveCodes.toArray(sortedEffectiveCodesArr);
+    // The REST API endpoint for "api.getAvailableRegions(commaSeparatedCodes)" accepts utmost
+    // 100 effective codes. So, sending the effective codes in batches of 100.
+    for (int minIdx = 0; minIdx < sortedEffectiveCodesArr.length; minIdx += MAX_EFFECTIVE_CODES_PER_API_CALL) {
+      int maxIdx = minIdx + MAX_EFFECTIVE_CODES_PER_API_CALL;
+      if (maxIdx > sortedEffectiveCodesArr.length) {
+        maxIdx = sortedEffectiveCodesArr.length - 1;
+      }
+      String startEffectiveCode = sortedEffectiveCodesArr[minIdx];
+      String endEffectiveCode = sortedEffectiveCodesArr[maxIdx];
+      TreeSet<String> subSetOfSortedEffectiveCodes = new TreeSet<String>(sortedEffectiveCodes.subSet(startEffectiveCode,
+              endEffectiveCode));
+      commaSeparatedCodes = getCommaSeparatedEffectiveCodes(subSetOfSortedEffectiveCodes);
+      response = api2.getAvailableRegions(commaSeparatedCodes);
+      responseIterator = response.iterator();
+      for (String effectiveCode : subSetOfSortedEffectiveCodes) {
+        Region region1 = regionGiven(effectiveCode, topLevelRegions);
+        Collection<Region> thirdLevelRegions = responseIterator.next();
+        region1.setChildRegions(thirdLevelRegions);
+      }
+    }
+
+    return topLevelRegions;
+  }
+
+  private TreeSet<Region> getSortedSecondLevelRegions(Collection<Region> topLevelRegions) {
+    TreeSet<Region> secondLevelRegions = new TreeSet<Region>();
+    for (Region topLevelRegion : topLevelRegions) {
+      secondLevelRegions.addAll(topLevelRegion.getChildRegions());
+    }
+    return secondLevelRegions;
+  }
+
+  private Region regionGiven(String effectiveCode, Collection<Region> regions) {
+    if (effectiveCode == null || regions == null) {
+      return null;
+    }
+    for (Region region : regions) {
+      if (region.getEffectiveCode().equals(effectiveCode)) {
+        return region;
+      }
+      TreeSet<Region> childRegions = region.getChildRegions();
+      Region region1 = regionGiven(effectiveCode, childRegions);
```

Comment:
Using digits in variable names like this is usually not preferred, please rename it to use some logical name instead. Take care of this in all the files.